src/wp-includes/load.php

```diff
@@ -1526,6 +1526,30 @@ function wp_is_jsonp_request() {
 
 }
 
+/**
+ * Checks whether a string is a valid JSON Content-Type.
+ *
+ * @since 5.6.0
+ *
+ * @staticvar bool $is_json
+ *
+ * @param string $content_type A Content-Type string
+ * @return bool True if string is a valid JSON Content-Type.
+ */
+function wp_is_json_content_type( $content_type ) {
+	static $is_json_content_type = array();
+
+	if ( ! isset( $is_json_content_type[ $content_type ] ) ) {
+		$is_json_content_type[ $content_type ] = (bool) preg_match( '/(^|\s|,)application\/([\w!#\$&-\^\.\+]+\+)?json(\+oembed)?($|\s|;|,)/i', $content_type );
+	}
+
+	if ( ! is_bool( $is_json_content_type[ $content_type ] ) ) {
```

Comment:
How would this branch get executed? The result of `preg_match` is cased to a boolean.

Reply:
It returns `1` if there is a match, `0` if there is none and `false` if there is an error, so casting to bool, should be fine.

Reply:
I agree casting to book is fine, but that means that the `is_bool` check should never return false, no?

Reply:
I agree, I removed it in one of my last commits.